app/models/provider/synth.rb

```diff
@@ -0,0 +1,59 @@
+class Provider::Synth < Provider::Base
+  def initialize(api_key)
+    @api_key = api_key
+  end
+
+  def fetch_exchange_rate(from:, to:, date:)
+    response = Faraday.get("#{base_url}/rates/historical") do |req|
+      req.headers["Authorization"] = "Bearer #{api_key}"
+      req.params["date"] = date.to_s
+      req.params["from"] = from
+      req.params["to"] = to
+    end
+
+    if response.success?
+      ExchangeRateResponse.new \
+        rate: JSON.parse(response.body).dig("data", "rates", to),
+        raw_response: response
+    else
+      raise Provider::Base::ProviderError, <<~ERROR
+        Failed to fetch exchange rate from #{self.class}
+          Status: #{response.status}
+          Response: #{response.body.inspect}
+      ERROR
+    end
```

Comment:
Why not just return the rate - what do you gain from returning a `ExchangeRateResponse`?